fix-standard-sql-regex

courtney-lang opened the pull request:
## Summary
Fix issue where a string following the same pattern (ex: 'abcd.efgh.ijkl') is interpreted as table, causing an error.

Issue https://github.com/servian/bigquery-view-analyzer/issues/9


### Standard Sql Regex Changes
- Make the first backtick at beginning of project name required instead of optional
  - Per [BigQuery DDL documentation](https://cloud.google.com/bigquery/docs/reference/standard-sql/data-definition-language#create_view_statement):
    >When you use a DDL statement to create a view, you must specify the project, dataset, and view in the following format: \`project_id.dataset.table\` (including the backticks); for example, \`myproject.mydataset.newview\`. 
  - Examples:
    - [Current regex](https://regex101.com/r/E9mK3B/1)
    - [Fixed regex](https://regex101.com/r/E9mK3B/2)

- Remove dash character from dataset & table qualifiers
  - Per [BigQuery Datasets documentation](https://cloud.google.com/bigquery/docs/datasets#dataset-naming):
    >Dataset names cannot: 
    >- Contain spaces or special characters such as -, &, @, or %
  - Per [BigQuery Tables documentation](https://cloud.google.com/bigquery/docs/tables#table_naming)
    >The table name can:
    >- Contain up to 1,024 characters
    >- Contain letters (upper or lower case), numbers, and underscores

### Test Changes
- Add test for invalid table references
- Add empty string (`""`) and `"INNER"` to join prefixes

christippett commented:
Good pick-up @courtney-lang, I hadn't anticipated that table pattern. Can I ask you to please add the affected table name to the tests before I merge? Thanks!

courtney-lang commented:
@christippett I separated out the 4 standard sql table references that fail with the regex changes into a separate test asserting they don't match. I also noticed that inner joins weren't being tested, not that it's super important, so I added an empty string and `INNER` to the join_prefixes.

Additionally, I removed the dash character `-` from dataset and table groups given the BigQuery documentation. Also not super important since the view can't even be created with a dash in the dataset or table name, but I figure being more explicit never hurts.

I updated the PR summary to be more detailed about the changes and included linked documentation.

christippett commented:
Thanks @courtney-lang for your help on this, I really appreciate it. I'll publish a new release today with your latest changes.

jared-martin commented:
@christippett any plans to publish this release (20.1.1) to pypi?

christippett commented:
Opps, I didn't notice the publish step in the build failed. It should be up on PyPI now.
